FreeRTOS/Demo/Posix_GCC/console.c

```diff
@@ -39,7 +39,15 @@ StaticSemaphore_t xStdioMutexBuffer;
 
 void console_init( void )
 {
-    xStdioMutex = xSemaphoreCreateMutexStatic( &xStdioMutexBuffer );
+    #if( configSUPPORT_STATIC_ALLOCATION == 1 )
+    {
+        xStdioMutex = xSemaphoreCreateMutexStatic( &xStdioMutexBuffer );
+    }
+    #else /* if( configSUPPORT_STATIC_ALLOCATION == 1 ) */
```

Comment:
We're assuming if `configSUPPORT_STATIC_ALLOCATION` is not 1, then `configSUPPORT_DYNAMIC_ALLOCATION` must be 1. And that's not correct. Is there any compiling check if both `configSUPPORT_STATIC_ALLOCATION` and `configSUPPORT_DYNAMIC_ALLOCATION` are 0 in this demo?

Reply:
Since we are using heap_3.c , if `configSUPPORT_STATIC_ALLOCATION` is not 1, it automatically suggests we are using dynamic allocation, that was the rationale behind this assumption.

`configSUPPORT_DYNAMIC_ALLOCATION` cannot be set to 0 in this demo.